server/src/main/java/org/opensearch/index/mapper/MatchOnlyTextFieldMapper.java

```diff
@@ -0,0 +1,310 @@
+            super(name, indexCreatedVersion, indexAnalyzers);
+        }
+
+        @Override
+        public MatchOnlyTextFieldMapper build(BuilderContext context) {
+            // TODO - disable norms and index-options and validate
+            FieldType fieldType = TextParams.buildFieldType(index, store, indexOptions, norms, termVectors);
+            MatchOnlyTextFieldType tft = buildFieldType(fieldType, context);
+            return new MatchOnlyTextFieldMapper(
+                name,
+                fieldType,
+                tft,
+                buildPrefixMapper(context, fieldType, tft),
+                buildPhraseMapper(fieldType, tft),
+                multiFieldsBuilder.build(this, context),
+                copyTo.build(),
+                this
+            );
+        }
+
+        @Override
+        protected MatchOnlyTextFieldType buildFieldType(FieldType fieldType, BuilderContext context) {
+            NamedAnalyzer indexAnalyzer = analyzers.getIndexAnalyzer();
+            NamedAnalyzer searchAnalyzer = analyzers.getSearchAnalyzer();
+            NamedAnalyzer searchQuoteAnalyzer = analyzers.getSearchQuoteAnalyzer();
+
+            if (fieldType.indexOptions().compareTo(IndexOptions.DOCS) > 0) {
+                throw new IllegalArgumentException("Cannot set position_increment_gap on field [" + name + "] without positions enabled");
+            }
+            if (positionIncrementGap.get() != POSITION_INCREMENT_GAP_USE_ANALYZER) {
+                if (fieldType.indexOptions().compareTo(IndexOptions.DOCS) < 0) {
+                    throw new IllegalArgumentException(
+                        "Cannot set position_increment_gap on field [" + name + "] without indexing enabled"
+                    );
+                }
+                // for index analyzer we don't set positionIncrementGap whereas for search analyzer its set because
+                // phrase queries, which make use of it, should work fine as they will directly work on the field value
+                // per matched document by reading from _source field.
+                indexAnalyzer = new NamedAnalyzer(indexAnalyzer, positionIncrementGap.get());
+                searchAnalyzer = new NamedAnalyzer(searchAnalyzer, positionIncrementGap.get());
+                searchQuoteAnalyzer = new NamedAnalyzer(searchQuoteAnalyzer, positionIncrementGap.get());
+            }
+            TextSearchInfo tsi = new TextSearchInfo(fieldType, similarity.getValue(), searchAnalyzer, searchQuoteAnalyzer);
+            MatchOnlyTextFieldType ft = new MatchOnlyTextFieldType(
+                buildFullName(context),
+                index.getValue(),
+                fieldType.stored(),
+                tsi,
+                meta.getValue()
+            );
+            ft.setIndexAnalyzer(indexAnalyzer);
+            ft.setEagerGlobalOrdinals(eagerGlobalOrdinals.getValue());
+            ft.setBoost(boost.getValue());
```

Comment:
For my understanding, do we make use of the `TextFieldMapper`'s builder to inherit these values? Because we do not set them locally here.

Reply:
we are calling `buildFieldType()` from `MatchOnlyTextFieldMappe`r builder `build()` above, if that's what you meant?